panda_controller:
  type: "velocity_controllers/JointTrajectoryController"
  joints:
    - joint1
    - joint2
    - joint3
    - joint4
    - joint5
    - joint6
    - joint7
    # - panda_finger_joint1
    # - panda_finger_joint2
  
  /gazebo_ros_control:
    pid_gains:
      joint1: { p: 600, d: 30, i: 0 }
      joint2: { p: 600, d: 30, i: 0 }
      joint3: { p: 600, d: 30, i: 0 }
      joint4: { p: 600, d: 30, i: 0 }
      joint5: { p: 250, d: 10, i: 0 }
      joint6: { p: 150, d: 10, i: 0 }
      joint7: { p: 50, d: 5, i: 0 }

  # gains:
  #   joint1: { p: 600, d: 30, i: 0 }
  #   joint2: { p: 600, d: 30, i: 0 }
  #   joint3: { p: 600, d: 30, i: 0 }
  #   joint4: { p: 600, d: 30, i: 0 }
  #   joint5: { p: 250, d: 10, i: 0 }
  #   joint6: { p: 150, d: 10, i: 0 }
  #   joint7: { p: 50, d: 5, i: 0 }

  constraints:
    #goal_time: 0.6
    #stopped_velocity_tolerance: 0.05
    joint1: {trajectory: 0.01, goal: 0.05}
    joint2: {trajectory: 0.01, goal: 0.05}
    joint3: {trajectory: 0.01, goal: 0.05}
    joint4: {trajectory: 0.01, goal: 0.05}
    joint5: {trajectory: 0.01, goal: 0.05}
    joint6: {trajectory: 0.01, goal: 0.05}
    joint7: {trajectory: 0.01, goal: 0.05}
    # finger_joint1: {trajectory: 0.05, goal: 0.05}
    # finger_joint2: {trajectory: 0.05, goal: 0.05}
  #stop_trajectory_duration: 0.5 
  state_publish_rate:  2500
  action_monitor_rate: 10
  min_vel_x: 0.0
  min_vel_y: 0.0
  min_vel_z: 0.0
  max_val_x: 0.5
  max_val_y: 0.5
  max_val_z: 0.5
  